docker/recorder/src/test/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironmentTest.java

```diff
@@ -131,4 +168,120 @@ public void should_report_schema_of_docker_composition() {
+        final TakeDockerEnvironment takeDockerEnvironment = new TakeDockerEnvironment();
+        takeDockerEnvironment.propertyReportEvent = propertyReportEvent;
+
+        takeDockerEnvironment.reportContainerStatsAfterTest(new After(TakeDockerEnvironmentTest.class, TakeDockerEnvironmentTest.class.getMethod("should_report_container_stats")), dockerClientExecutor, cubeRegistry);
+        verify(propertyReportEvent).fire(propertyReportEventArgumentCaptor.capture());
+
+        final PropertyReportEvent propertyReportEvent = propertyReportEventArgumentCaptor.getValue();
+        final PropertyEntry propertyEntry = propertyReportEvent.getPropertyEntry();
+        assertThat(propertyEntry).isInstanceOf(GroupEntry.class);
+
+        GroupEntry parent = (GroupEntry) propertyEntry;
+        final List<PropertyEntry> rootEntries = parent.getPropertyEntries();
+        assertThat(rootEntries).hasSize(1);
+
+        List<PropertyEntry> entryList = rootEntries.get(0).getPropertyEntries();
+
+        assertThat(entryList).hasSize(3).extracting("class.simpleName").containsExactly("GroupEntry", "GroupEntry", "GroupEntry");
+        assertThat(entryList).extracting("name").contains("network statistics", "memory statistics", "block I/O statistics");
+
+
+        PropertyEntry nw = entryList.get(0);
+        assertThat(nw).isInstanceOf(GroupEntry.class);
+        List<PropertyEntry> nwList = nw.getPropertyEntries();
+
+        assertThat(nwList).hasSize(2).extracting("class.simpleName").containsExactly("GroupEntry", "GroupEntry");
+        assertThat(nwList).flatExtracting("propertyEntries").containsExactly(
+                new KeyValueEntry("rx_bytes","724 B"),
+                new KeyValueEntry("tx_bytes", "418 B"),
+                new KeyValueEntry("rx_bytes", "724 B"),
+                new KeyValueEntry("tx_bytes", "418 B"));
+
+        PropertyEntry memory = entryList.get(1);
+        assertThat(memory).isInstanceOf(GroupEntry.class);
+        List<PropertyEntry> memList = memory.getPropertyEntries();
+
+        assertThat(memList).hasSize(3).extracting("class.simpleName").containsExactly("GroupEntry", "GroupEntry", "GroupEntry");
+        assertThat(memList).flatExtracting("propertyEntries").containsExactly(
+                new KeyValueEntry("usage", "33.5 MiB"),
+                new KeyValueEntry("max_usage", "34.1 MiB"),
+                new KeyValueEntry("limit", "19.0 GiB"));
+
+        PropertyEntry blkIO = entryList.get(2);
+        assertThat(blkIO).isInstanceOf(GroupEntry.class);
+        List<PropertyEntry> blkIOList = blkIO.getPropertyEntries();
+
+        assertThat(blkIOList).hasSize(2).extracting("class.simpleName").containsExactly("GroupEntry", "GroupEntry");
+        assertThat(blkIOList).flatExtracting("propertyEntries").containsExactly(
+                new KeyValueEntry("I/O Bytes Read", "49.5 KiB"),
+                new KeyValueEntry("I/O Bytes Write", "0 B"));
+    }
```

Comment:
This test validates 3 functionalities at once. How about 3 separated, focused test methods instead?